File: 013. Roman to Integer/solution.py
class Solution:
    def romanToInt(self, s):
        """
        :type s: str
        :rtype: int
        """
        mapp = {
            'I':1,
            'V':5,
            'X':10,
            'L':50,
            'C':100,
            'D':500,
            'M':1000,
        }
        
        ans = mapp[s[-1]]
        last_strs = s[-1]
        for strs in s[-2::-1]:
            if mapp[strs] < mapp[last_strs]:
                ans = ans - mapp[strs]
            else:
                ans = ans + mapp[strs]
            last_strs = strs
        return ans

File: 013. Roman to Integer/solution.cpp
class Solution {
public:
    int romanToInt(string s) {
        map<char,int> mapp;
        mapp.insert(make_pair<char,int>('I',1));
        mapp.insert(make_pair<char,int>('V',5));
        mapp.insert(make_pair<char,int>('X',10));
        mapp.insert(make_pair<char,int>('L',50));
        mapp.insert(make_pair<char,int>('C',100));
        mapp.insert(make_pair<char,int>('D',500));
        mapp.insert(make_pair<char,int>('M',1000));
        
        int sum =mapp[s[s.length()-1]];
        for (int i = s.length()-2 ; i >= 0 ; i --){
            if(mapp[s[i]] < mapp[s[i+1]]){
                sum -= mapp[s[i]];
            }
            else{
                sum += mapp[s[i]];
            }
        }
        
        return sum;
        
    }
};